### PR TITLE
Media Card Seek Functionality

### LINKED ISSUE #4493: Seeking capabilities for media player card
**Is your feature request related to a problem? Please describe.**

Our media players entities can implement seek capabilities as part of their supported features, as seen by the small yellow/orange/ish bar in the media player card:

image

However, there is no way to seek 🤷‍♂ 

There is a service call, but that feels rather useless (unless you want a song to start on a specific timestamp from an automation or something).

**Describe the solution you'd like**

My wish for this card would be that the yellow/orange/ish bar is clickable like a media player, allowing me to seek through the media.

**Describe alternatives you've considered**

🤷‍♂

**Additional context**

I've been reworking the Spotify integration in:
<https://github.com/home-assistant/home-assistant/pull/30717>

I've managed to implement the current playing position and seeking capabilities.
When I finally implemented it... I was like, wait uh? Does it work? 😂

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Ability to seek using the Progress Bar

[Screenshot from Gyazo](https://gyazo.com/dda60d4a5f078bd6ce43cd2c7fd5f3c1)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

- This PR fixes or closes issue: fixes #4493 
- Link to documentation pull request: TODO

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 1]
```diff
@@ -383,6 +394,10 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
         --paper-progress-active-color: var(--accent-color);
         --paper-progress-container-color: rgba(200, 200, 200, 0.5);
       }
+
+      .progress:hover {
+        --paper-progress-height: 8px;
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: I am assuming this isn't the best way to do this. I wanted to make sure the user knows that they are on the progress bar. So updating the height would show them clicking will perform an action.

I tried using a calculation of the var height * 2 but that didn't seem to work

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 2]
```diff
@@ -273,6 +274,16 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
     });
   }
 
+  private _handleSeek(e: MouseEvent): void {
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Check for `SUPPORT_SEEK` <https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/const.py#L50>

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: I think it is fine like this, you could add a new var if you really wanted...

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 3]
```diff
@@ -119,6 +119,7 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
                 .max="${stateObj.attributes.media_duration}"
                 .value="${stateObj.attributes.media_position}"
                 class="progress"
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Add a classMap that adds a class when it supports `SEEK` and only add the hover height in that class.

**Comment by homeassistant** [comment]
Hello @zsarnett,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`09439abc01de62967a8fe78f2a10cd0f2389fe45`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant-polymer&#x2F;pull&#x2F;4907&#x2F;commits&#x2F;09439abc01de62967a8fe78f2a10cd0f2389fe45) - No email found attached to the commit.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Comment by homeassistant** [comment]
Hello @zsarnett,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`2fa236508272922aef4070533e0abc28421fd1d9`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant-polymer&#x2F;pull&#x2F;4907&#x2F;commits&#x2F;2fa236508272922aef4070533e0abc28421fd1d9) - No email found attached to the commit.
- [`09439abc01de62967a8fe78f2a10cd0f2389fe45`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant-polymer&#x2F;pull&#x2F;4907&#x2F;commits&#x2F;09439abc01de62967a8fe78f2a10cd0f2389fe45) - No email found attached to the commit.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Review by bramkragten — APPROVED** [review]

**Pull request merged**